plugins/action/zos_unarchive.py

```diff
@@ -0,0 +1,112 @@
+
+        if module_args.get("remote_src", False):
+            result.update(
+                self._execute_module(
+                    module_name="ibm.ibm_zos_core.zos_unarchive",
+                    module_args=module_args,
+                    task_vars=task_vars,
+                )
+            )
+        else:
+            source = module_args.get("path") if module_args.get("path") is not None else module_args.get("src")
+            force = module_args.get("force")
+            format = self._task.args.get("format")
+            format_name = format.get("name")
+            copy_module_args = dict()
+            dest_data_set = dict()
+            dest = ""
+            if source.startswith('~'):
+                source = os.path.expanduser(source)
+            source = os.path.realpath(source)
+
+            if format_name in USS_SUPPORTED_FORMATS:
+                dest = self._execute_module(
+                    module_name="tempfile", module_args={}, task_vars=task_vars,
+                ).get("path")
+            elif format_name in MVS_SUPPORTED_FORMATS:
+                tmp_hlq = module_args.get("tmp_hlq") if module_args.get("tmp_hlq") is not None else ""
+                cmd_res = self._execute_module(
+                    module_name="command",
+                    module_args=dict(
+                        _raw_params="mvstmp {0}".format(tmp_hlq)
+                    ),
+                    task_vars=task_vars,
+                )
+                dest = cmd_res.get("stdout")
+                if format_name == 'terse':
+                    dest_data_set = dict(type='SEQ', record_format='FB', record_length=1024)
+                if format_name == 'xmit':
+                    dest_data_set = dict(type='SEQ', record_format='FB', record_length=80)
+            else:
+                # Raise unsupported format name
+                None
+
+            copy_module_args.update(
+                dict(
+                    src=source,
+                    dest=dest,
+                    dest_data_set=dest_data_set,
+                    force=force,
+                    # is_binary=True,
+                )
+            )
+            copy_task = copy.deepcopy(self._task)
+            copy_task.args = copy_module_args
+            zos_copy_action_module = ZosCopyActionModule(task=copy_task,
+                                                         connection=self._connection,
+                                                         play_context=self._play_context,
+                                                         loader=self._loader,
+                                                         templar=self._templar,
+                                                         shared_loader_obj=self._shared_loader_obj)
+            result.update(zos_copy_action_module.run(task_vars=task_vars))
+
+            module_args["path"] = dest
+            display.vvv(u"Copy args {0}".format(result), host=self._play_context.remote_addr)
```

Comment:
Have you run the module with Ansible 2.15? I think it could trigger the deprecation warning from #779.

Reply:
You're right will extract the fix from #806

Reply:
Actually not sure, since is not using _play_context.verbosity ... let me test and get back

Reply:
You're right, and I'm guessing you tested and didn't find any warnings.